### PR TITLE
Refactor NewDetector for application drift detector

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
Fix for https://github.com/pipe-cd/pipecd/issues/3156

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nghialv** [comment]
Nice. Thank you.
/lgtm

**Comment by knanao** [comment]
@nghialv 
Sorry, I fixed again a little bit.
ptal when you have some time🙇

**Comment by nakabonne** [comment]
Thank you
/lgtm

File: pkg/app/piped/driftdetector/detector.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ type secretDecrypter interface {
 
 type Detector interface {
 	Run(ctx context.Context) error
+	ReportApplicationSyncState(ctx context.Context, appID string, state model.ApplicationSyncState) error
```
Comment (nghialv) [inline pkg/app/piped/driftdetector/detector.go:62]: Why do we need to add this function?

Comment (knanao) [reply pkg/app/piped/driftdetector/detector.go]: As you say, it's not currently in use. Hence I removed it. TYVM:)

Comment (nghialv) [reply pkg/app/piped/driftdetector/detector.go]: Thank you.

**Comment by nghialv** [comment]
Nice. Keep up your great work.
/approve

**Pull request merged**